docs: specify case of hex-encoded string

lucab opened the pull request:
This updates the `random_bytes.hex` docs in order to specify that the hex-encoded string uses the lowercase alphabet. It also adds a note about the fixed size of the result.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform-provider-random?pullRequest=502)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform-provider-random?pullRequest=502)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform-provider-random?pullRequest=502).</sub>

bendbennett commented on docs/resources/bytes.md:
```diff
@@ -37,8 +37,8 @@ resource "azurerm_key_vault_secret" "jwt_secret" {
-- `base64` (String, Sensitive) The generated bytes presented in base64 string format.
-- `hex` (String, Sensitive) The generated bytes presented in hex string format.
+- `base64` (String, Sensitive) The generated bytes presented in base64 format.
+- `hex` (String, Sensitive) The generated bytes presented in lowercase hex format. The length of the encoded string is exactly twice the `length` parameter.
```
```suggestion
- `hex` (String, Sensitive) The generated bytes presented in lowercase hexadecimal format. The length of the encoded string is exactly twice the `length` parameter.
```

bendbennett reviewed (commented):
Hi @lucab 👋 

Thank you for your PR.
I've made one small suggestion on the wording.

lucab commented:
@bendbennett ack thanks, directly committed your suggestion.

bendbennett reviewed (commented):
Hi @lucab 

I forgot to mention that'll you need to make the changes to the wording on the `Schema` and then run `make generate` to recreate the docs.

lucab commented:
@bendbennett my bad, I didn't realize the docpage was autogenerate, I totally missed the very clear header, apologies.
I've now reworked and rebased the PR with code changes, then regenerate the docs.

bendbennett reviewed (approved):
LGTM 🚀 

Thank you for the contribution @lucab
